x/campaign/keeper/msg_server_burn_vouchers_test.go

```diff
@@ -0,0 +1,134 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	spnerrors "github.com/tendermint/spn/pkg/errors"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/campaign/types"
+)
+
+func TestMsgBurnVouchers(t *testing.T) {
+	var (
+		campaignKeeper, _, _, bankKeeper, campaignSrv, _, sdkCtx = setupMsgServer(t)
+
+		campaign       = sample.Campaign(0)
+		ctx            = sdk.WrapSDKContext(sdkCtx)
+		addr           = sample.Address()
+		vouchersTooBig = sdk.NewCoins(
+			sdk.NewCoin("v/0/foo", sdk.NewInt(types.DefaultTotalShareNumber+1)),
+		)
+	)
+
+	// Create shares
+	shares, err := types.NewShares("1000foo,500bar,300foobar")
+	require.NoError(t, err)
+
+	// Set campaign
+	campaign.AllocatedShares = shares
+	campaign.Id = campaignKeeper.AppendCampaign(sdkCtx, campaign)
+
+	// Create vouchers
+	vouchers, err := types.SharesToVouchers(shares, campaign.Id)
+	require.NoError(t, err)
+
+	// Send coins to account
+	err = bankKeeper.MintCoins(sdkCtx, types.ModuleName, vouchers)
+	require.NoError(t, err)
+	err = bankKeeper.SendCoinsFromModuleToAccount(sdkCtx, types.ModuleName, addr, vouchers)
+	require.NoError(t, err)
+
+	for _, tc := range []struct {
+		name string
+		msg  types.MsgBurnVouchers
+		err  error
+	}{
+		{
+			name: "non existing campaign",
+			msg: types.MsgBurnVouchers{
+				Creator:    addr.String(),
+				CampaignID: 1000,
+				Vouchers:   sample.Coins(),
+			},
+			err: types.ErrCampaignNotFound,
+		},
+		{
+			name: "invalid creator address",
+			msg: types.MsgBurnVouchers{
+				Creator:    "invalid_address",
+				CampaignID: campaign.Id,
+				Vouchers:   sample.Coins(),
+			},
+			err: spnerrors.ErrCritical,
+		},
+		{
+			name: "should not burn more than allocated shares",
+			msg: types.MsgBurnVouchers{
+				Creator:    addr.String(),
+				CampaignID: campaign.Id,
+				Vouchers:   vouchersTooBig,
+			},
+			err: spnerrors.ErrCritical,
+		},
+		{
+			name: "burn vouchers",
```

Comment:
We should test this message several time for the same campaign and test the edge case once all vouchers are burned

Reply:
we already have a [test for all burned vouchers](https://github.com/tendermint/spn/pull/329/files#diff-dc5083900b9c760a253d327bac3c61ab00573fe77a5ba660fa5d733e9f127dbaR84-R90), this `burn vouchers` test, burn all vouchers, and the seconds try to burn again, and it's will fails

Reply:
I meant we should burn some and then try to burn again on the same campaign

Reply:
added on https://github.com/tendermint/spn/pull/329/commits/7dea6680a2e389986c22ae538b1c7051cf19b24d